libweasyl/oauth.py

```diff
@@ -155,3 +181,78 @@ def revoke_consumers_for_user(userid, clientids):
         .filter_by(userid=userid)
         .filter(OAuthBearerToken.clientid.in_(clientids)))
     q.delete('fetch')
+
+
+def register_client(userid, name, scopes, redirects, homepage):
+    """
+    Register an application as an OAuth2 consumer
+    :param userid: the user registering this application
+    :param name: the name of the application
+    :param scopes: a list of the scopes registered for this application
+    :param redirects: allowed redirect URIs for this application
+    """
+
+    session = OAuthConsumer.dbsession
+    new_consumer = OAuthConsumer(
+        clientid=security.generate_key(32),
```

Comment:
This one doesn't need to be secure, right? Should it be more meaningful? Knowing the userid for example might be useful.